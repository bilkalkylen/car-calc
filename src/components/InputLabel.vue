<script setup lang="ts">
export interface Props {
  text: string;
  tag?: string;
}

withDefaults(defineProps<Props>(), { tag: 'label' });
</script>

<template>
  <component :is="tag" class="input-label">
    {{ text }}
    <slot />
  </component>
</template>

<style lang="scss" scoped>
.input-label {
  font-weight: bold;
  font-size: 0.8rem;
  font-style: italic;

  @media screen and (min-width: $size-small-tablet) {
    font-size: 0.9rem;
  }

  @media screen and (min-width: $size-tablet) {
    font-size: 1rem;
  }
}
</style>
