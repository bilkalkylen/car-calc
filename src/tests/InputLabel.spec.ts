import { render } from '@testing-library/vue';
import { describe, expect, it } from 'vitest';

import InputLabel from '@/components/InputLabel.vue';

describe('InputLabel', () => {
  it('renders correctly', () => {
    const { container } = render(InputLabel, { props: { text: 'label text' } });

    expect(container).toMatchSnapshot();
  });

  it('renders the correct tag when passed', () => {
    const screen = render(InputLabel, { props: { text: "I'm a div", tag: 'div' } });

    expect(screen.getByText("I'm a div").tagName).toBe('DIV');
  });
});
