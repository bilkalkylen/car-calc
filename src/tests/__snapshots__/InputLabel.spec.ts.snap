// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InputLabel > renders correctly 1`] = `
<div>
  <label
    class="input-label"
    data-v-4c048e95=""
  >
    label text 
    
    
  </label>
</div>
`;
